File: _scala/primer/tuples.scala
// NOTE: tuples are immutable
val tuple_one = (1.0, "String", 4)              // (1.0,String,4)
val tuple_two = ("Alpha", "Bravo", (1, 0))      // (Alpha,Bravo,(1,0))

// get value
tuple_one._2                                    // String
tuple_two._3._2                                 // 0

// assign each value in tuple to a variable
val (first, second, third) = ("Alpha", "Beta", "Gamma")
first                                           // Alpha
second                                          // Beta
third                                           // Gamma

File: python/_ignore/primer/tuples.py
# 2018-06

# tuples are immutable
tuple_one = (1.0, "String", 4)
tuple_two = ("Alpha", "Bravo", (1, 0))

print(tuple_one)
# (1.0, 'String', 4)
print(tuple_two)
# ('Alpha', 'Bravo', (1, 0))
print(tuple_two[2][1])
# 0

# concatenate
tuples = tuple_one + tuple_two
print(tuples)
# (1.0, 'String', 4, 'Alpha', 'Bravo', (1, 0))

# list to tuple
print(tuple([100, 'B']))
# (100, 'B')